libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/route/FasterRouteDetector.java

```diff
@@ -0,0 +1,135 @@
+package com.mapbox.services.android.navigation.v5.route;
+
+import android.location.Location;
+
+import com.mapbox.api.directions.v5.models.DirectionsResponse;
+import com.mapbox.api.directions.v5.models.DirectionsRoute;
+import com.mapbox.api.directions.v5.models.LegStep;
+import com.mapbox.api.directions.v5.models.RouteLeg;
+import com.mapbox.services.android.navigation.v5.routeprogress.RouteProgress;
+import com.mapbox.services.android.navigation.v5.routeprogress.RouteStepProgress;
+import com.mapbox.services.android.navigation.v5.utils.time.TimeUtils;
+
+import java.util.Date;
+import java.util.concurrent.TimeUnit;
+
+import retrofit2.Response;
+
+import static com.mapbox.services.android.navigation.v5.navigation.NavigationConstants.NAVIGATION_CHECK_FASTER_ROUTE_INTERVAL;
+import static com.mapbox.services.android.navigation.v5.navigation.NavigationConstants.NAVIGATION_MEDIUM_ALERT_DURATION;
+
+public class FasterRouteDetector extends FasterRoute {
+
+  private static final int VALID_ROUTE_DURATION_REMAINING = 600;
+
+  private Location lastCheckedLocation;
+
+  @Override
+  public boolean shouldCheckFasterRoute(Location location, RouteProgress routeProgress) {
```

Comment:
@mapbox/navigation-ios When you have a chance, can I get some 👀  on `shouldCheckFasterRoute` and `isFasterRoute`?  The former determines when a new route request should fire, while the latter determines if the `DirectionsRoute` from the response is actually faster the current route.  Thanks!